### DISCUSSION TITLE
useAnimatedGestureHandler: Difference between onEnd and onFinish

### DISCUSSION THREAD

**Discussion by friedebold** [software-mansion/react-native-reanimated#1949] [Q&A]
I don't seem to fully understand the difference between `onEnd` and `onFinish` when using `useAnimatedGestureHandler()`.

`onFinish` seems to be triggered at the same time as `onEnd` with some additional triggers when no animated value is changed.

Was wondering if someone knows since the documentation is not fully descriptive.

**Comment by FreTimmerman** [comment]
i want to know this too. neither are documented.

the only difference i found, was that on android, a "tap" -without actually dragging- fires `onFinish`, but not `onEnd`. while actually dragging fires both `onEnd` and `onFinish`.

weirdly on iOS a tap fires the same as a drag
